### PR TITLE
rt-kernel: Implement pnal_eth_get_status()

### PR DESCRIPTION
**Pull request by fredrikdanielmoller**

### CODE REVIEW

**Review by pyhys — APPROVED** [review]
Great work!
I think @hefloryd needs to look at the details.

File: src/ports/rt-kernel/dwmac1000.c [hunk 1]
```diff
@@ -280,6 +297,68 @@ static void dwmac1000_pbuf_dump (struct pbuf * p)
...
+   }
+}
+
+/**
+ * Calculate advertised capabilites
+ *
+ * @param anar       In:    Contents of the ANAR register read from PHY.
+ * @return Profinet advertised capabilites as a bitfield.
+ */
+static uint16_t dwmac1000_advertised_capabilities (uint16_t anar)
```
Comment (hefloryd) [inline src/ports/rt-kernel/dwmac1000.c:333]: Conversion to PNAL types should preferrably happen in pnal.c. Ideally the ethernet driver should not need to know about profinet.

Comment (fredrikdanielmoller) [reply src/ports/rt-kernel/dwmac1000.c]: Moved conversion to PNAL types to pnal.c.

**Comment by fredrikdanielmoller** [comment]
Moved conversion to PNAL types to pnal.c.
No rebase done as to simplify diffing.

File: src/ports/rt-kernel/dwmac1000.c (hunk 1 above)
Comment (fredrikdanielmoller) [reply src/ports/rt-kernel/dwmac1000.c]: Done.

**Pull request merged**